src/pudl/metadata/fields.py

```diff
@@ -1290,6 +1290,10 @@
         "type": "string",
         "description": "The code representing the most predominant type of energy that fuels the generator.",
     },
+    "energy_source_code_1_plant_gen": {
+        "type": "string",
+        "description": "Associated plant_gens' code representing the most predominant type of energy that fuels the generator.",
+    },
```

Comment:
Is "Associated plant_gens" a reference to the association table? It's a little bit confusing if so. Maybe we could add a phrase at the end saying something like "Describes the `record_id_eia_plant_gen` value in `out_eia__yearly_assn_plant_parts_plant_gen` instead of this phrase at the beginning.

Reply:
all of these new columns only show up in this association table. So this description will only ever show up with the `out_eia__yearly_assn_plant_parts_plant_gen` table. So i'm not sure adding the table name here is helpful - we don't do that anywhere else.

I am not giddy about the description. I do like your suggestion here and otherwise to use `record_id_eia_plant_gen` instead of `plant_gen` for consistency. Is it the word "associated" that is throwing you off here? Are these better?

1. `record_id_eia_plant_gen`'s code representing the most predominant type of energy that fuels the generator.
2. Code representing the most predominant type of energy that fuels the `record_id_eia_plant_gen`'s generator.

Reply:
I like number 2!

Yeah, the word associated was throwing me off a bit. If you wanted to make it even simpler, you could just have the same definition as the non `plant_gen` column and then add the sentence: "Associated with the `record_di_eia_plant_gen` record at the end?